### PR TITLE
global: test coverage and docs for loans

### PR DESCRIPTION
**Pull request by BadrAly**
* BETTER increase test coverage and add docs for loans.

Signed-off-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: rero_ils/modules/items/api.py [hunk 1]
```diff
@@ -397,7 +396,7 @@ def last_location_pid(self):
 
         of the last loan.
         """
-        loan_location_pid = get_last_transaction_location_for_item(self.pid)
+        loan_location_pid = get_last_transaction_loc_for_item(self.pid)
```
Comment (jma) [inline rero_ils/modules/items/api.py]: why changing the name as it is less explicit

File: rero_ils/modules/loans/api.py [hunk 2]
```diff
@@ -183,24 +139,19 @@ def get_request_by_item_pid_by_patron_pid(item_pid, patron_pid):
 
 
 def get_loans_by_patron_pid(patron_pid):
-    """Return all checkout for patron."""
-    if not patron_pid:
-        raise CirculationException('Patron PID not specified')
+    """Return all loans for patron."""
```
Comment (jma) [inline rero_ils/modules/loans/api.py]: Does it returns also requests?

File: rero_ils/modules/loans/api.py [hunk 3]
```diff
@@ -183,24 +139,19 @@ def get_request_by_item_pid_by_patron_pid(item_pid, patron_pid):
...
         .params(preserve_order=True)\
         .filter('term', patron_pid=patron_pid)\
         .sort({'transaction_date': {'order': 'asc'}})\
         .scan()
-
     for loan in results:
         yield Loan.get_record_by_pid(loan.loan_pid)
 
 
-def get_last_transaction_location_for_item(item_pid):
```
Comment (jma) [inline rero_ils/modules/loans/api.py]: same as previous

File: rero_ils/modules/loans/listener.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""."""
+"""Listener for loans."""
```
Comment (jma) [inline rero_ils/modules/loans/listener.py]: Can be more verbose such as: Signals connector for Loan.

File: rero_ils/modules/items/api.py (hunk 1 above)
Comment (BadrAly) [reply rero_ils/modules/items/api.py]: very long name ( for pep8)

File: rero_ils/modules/loans/api.py (hunk 2 above)
Comment (BadrAly) [reply rero_ils/modules/loans/api.py]: yes.

File: rero_ils/modules/loans/listener.py (hunk 4 above)
Comment (BadrAly) [reply rero_ils/modules/loans/listener.py]: done

**Pull request merged**